hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCreate.java

```diff
@@ -146,6 +148,30 @@ public void testCreateNonRecursive2() throws Exception {
     assertIsFile(fs, testFile);
   }
 
+  @Test
+  public void testCreateOnRoot() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    Path testFile = path(AbfsHttpConstants.ROOT_PATH);
+    try {
```

Comment:
you shouldn't need the double catch. intercept() will return the caught exception, type case to the class of arg 1, so can be asserted on. and if an exception is not the one expected, the stack trace is *too important to lose*. so rethrow it or use as the cause of an assertion error
```java
AbfsRestOperationException e = intercept(...)
if (e.getStatusCode()!=HTTP_CONFLICT) {
 // rethrow if its not the expected one.
 throw e;
}

Reply:
Taken